pkg/auth/requests/authenticate.go

```diff
@@ -158,15 +157,47 @@ func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResp
 	authResp.User = token.UserID
 	authResp.UserPrincipal = token.UserPrincipal.Name
 	authResp.Groups = groups
-	logrus.Debugf("Auth filter calling provider to get extra info %v", token.AuthProvider)
+	authResp.Extras = getUserExtraInfo(token, u, attribs)
+	logrus.Debugf("Extras returned %v", authResp.Extras)
 
-	extraFromProvider := providers.GetUserExtraAttributes(token.AuthProvider, token)
-	if extraFromProvider != nil {
-		authResp.Extras = extraFromProvider
-	}
 	return authResp, nil
 }
 
+func getUserExtraInfo(token *v3.Token, u *v3.User, attribs *v3.UserAttribute) map[string][]string {
+	var readFromToken bool
+	extraInfo := make(map[string][]string)
+
+	if attribs != nil && attribs.Extra != nil && len(attribs.Extra) != 0 {
+		if token.AuthProvider == "" {
+			//gather all extraInfo for all external auth providers present in the userAttributes
+			for _, extra := range attribs.Extra {
+				for key, value := range extra {
+					extraInfo[key] = append(extraInfo[key], value...)
+				}
+			}
+		} else {
+			//authProvider is set in token
```

Comment:
How does this handle upgrades where all tokens were set to `local`?

Reply:
So yeah I think the original idea where if "local" we write down all provider's  info, that will handle the case where all kubeconfig tokens were set to "local". Also for this feature, external IDP value is the goal, not the local values. But we can set all. Thoughts?